test/FunctionalTests/Microsoft.OData.Core.Tests/Json/TranscodingWriteStreamTests.cs

```diff
@@ -0,0 +1,122 @@
+// </copyright>
+//---------------------------------------------------------------------
+
+#if NETCOREAPP3_1_OR_GREATER
+using System;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Text.Encodings.Web;
+using System.Threading.Tasks;
+using Microsoft.OData.Json;
+using Xunit;
+
+namespace Microsoft.OData.Tests.Json
+{
+    public class TranscodingWriteStreamTests
+    {
+        private const int BufferSize = 4 * 1024;
+
+        public static TheoryData WriteAsyncInputLatin =>
+            GetLatinTextInput(BufferSize, BufferSize);
+
+        public static TheoryData WriteAsyncInputUnicode =>
+            GetUnicodeText(BufferSize);
+
+        [Theory]
+        [MemberData(nameof(WriteAsyncInputLatin))]
+        [MemberData(nameof(WriteAsyncInputUnicode))]
+        public Task WriteAsync_Works_WhenOutputIs_UTF32(string message)
+        {
+            Encoding targetEncoding = Encoding.UTF32;
+            return WriteAsyncTest(Encoding.UTF8, targetEncoding, message);
+        }
+
+        [Theory]
+        [MemberData(nameof(WriteAsyncInputLatin))]
+        [MemberData(nameof(WriteAsyncInputUnicode))]
+        public Task WriteAsync_Works_WhenOutputIs_Unicode(string message)
+        {
+            Encoding targetEncoding = Encoding.Unicode;
+            return WriteAsyncTest(Encoding.UTF8, targetEncoding, message);
+        }
+
+        [Theory]
+        [MemberData(nameof(WriteAsyncInputLatin))]
+        public Task WriteAsync_Works_WhenOutputIs_WesternEuropeanEncoding(string message)
+        {
+            // Arrange
+            Encoding targetEncoding = Encoding.GetEncoding(28591);
+            return WriteAsyncTest(Encoding.UTF8, targetEncoding, message);
+        }
+
+
+        [Theory]
+        [MemberData(nameof(WriteAsyncInputLatin))]
+        public Task WriteAsync_Works_WhenOutputIs_ASCII(string message)
+        {
+            // Arrange
+            Encoding targetEncoding = Encoding.ASCII;
+            return WriteAsyncTest(Encoding.UTF8, targetEncoding, message);
+        }
+
+        private static async Task WriteAsyncTest(Encoding inputEncoding, Encoding targetEncoding, string message)
+        {
+            string expected = JavaScriptEncoder.Default.Encode(message);
+
+            var stream = new MemoryStream();
+
+            using var transcodingStream = new TranscodingWriteStream(stream, targetEncoding, inputEncoding);
+            await transcodingStream.WriteAsync(inputEncoding.GetBytes(JavaScriptEncoder.Default.Encode(message)), default);
+            // The transcoding streams use Encoders and Decoders that have internal buffers. We need to flush these
+            // when there is no more data to be written. Stream.FlushAsync isn't suitable since it's
+            // acceptable to Flush a Stream (multiple times) prior to completion.
+            //await transcodingStream.FinalWriteAsync(default);

```

Comment:
```suggestion
            // await transcodingStream.FinalWriteAsync(default);
```

Reply:
Removing this entire comment as it's pointless here.